src/AutoMerge/guidelines.jl

```diff
@@ -331,6 +331,40 @@ function meets_standard_initial_version_number(version)
     end
 end
 
+function meets_has_license(data::GitHubAutoMergeData)
+    body = data.pr.body
+
+    if body === nothing
+        return false, "Something went wrong: pull request body is `nothing`"
+    end
+
+    repo, version, commit = repo_version_commit_of_pull_request_body(repo; skip_if_JLL_in_body=false)
+    repo === nothing && return false, "Could not parse repository from PR body"
+    version === nothing && return false, "Could not parse version from PR body"
+    commit === nothing && return false, "Could not parse commit from PR body"
+
+    files, pages = try
+        GitHub.directory(repo, ""; auth=data.auth, ref=commit)
```

Comment:
IIRC the API has a license endpoint. Perhaps use that and let GitHub find the file instead of trying the filenames below?

Reply:
Yeah, that was suggested in https://github.com/JuliaRegistries/RegistryCI.jl/issues/261#issuecomment-720518808 too. I think I'd prefer to use @DilumAluthge's other suggestion there and use `Pkg.add` to get the code instead of the GitHub API, to allow non-github packages.

I don't really see the advantage to using GitHub's license detection for github packages and a hardcoded list for non-github packages; is that it that GitHub packages are the common case and the license detection will have better chances at finding the license?

Reply:
I now use the code from Pkg.add, and use `licensecheck` to identify the license (discussed more in the OP)